crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,840 @@
+}
+
+enum StakeStatus {
+  PENDING
+  ACTIVE
+  UNSTAKED
+}
+
+type CoinMetadata {
+  decimals: Int
+  name: String
+  symbol: String
+  description: String
+  iconURL: String
+  supply: U64
+
+  asMoveObject: MoveObject
+}
+
+input DynamicFieldName {
+  type: TypeInst!
+  bcs: Bytes!
+}
+
+type DynamicField {
+  name: MoveValue
+  value: DynamicFieldValue
+}
+
+union DynamicFieldValue = MoveObject | MoveValue
+
+type MoveObject {
+  contents: MoveValue
+  hasPublicTransfer: Boolean
+
+  asCoin: Coin
+  asStakedSui: StakedSui
+  asObject: Object
+}
+
+type MovePackage {
+  modules: MovePackageModuleConnection
+
+  normalizedModules: NormalizedModulesConnection
+  normalizedModule(name: String!): NormalizedModule
+
+  asObject: Object
+}
+
+type MoveModuleId {
+  package: MovePackage
+  name: String
+}
+
+enum MoveAbility {
+  COPY
+  DROP
+  STORE
+  KEY
+}
+
+enum MoveVisibility {
+  PUBLIC
+  PRIVATE
+  FRIEND
+}
+
+enum MoveReference {
+  IMMUT
+  MUT
+}
+
+type MoveStructTypeParameterDecl {
+  constraints: [MoveAbility]
+  isPhantom: Boolean
+}
+
+type MoveFunctionTypeParameterDecl {
+  constraints: [MoveAbility]
+}
+
+type NormalizedModule {
+  fileFormatVersion: Int
+
+  moduleId: MoveModuleId
+  friends: [MoveModuleId]
+
+  structs: NormalizedStructConnection
+  struct(name: String!): NormalizedStruct
+
+  functions: NormalizedFunctionConnection
+  function(name: String!): NormalizedFunction
+}
+
+type NormalizedStruct {
+  module: NormalizedModule
+  name: String
+
+  abilities: [MoveAbility]
+  typeParameters: [MoveStructTypeParameterDecl]
+  fields: [NormalizedField]
+}
+
+type NormalizedField {
+  name: String
+  type: MoveTypeUse
+}
+
+type NormalizedFunction {
+  module: NormalizedModule
+  name: String
+
+  visibility: MoveVisibility
+  isEntry: Boolean
+
+  typeParameters: [MoveFunctionTypeParameterDecl]
+  parameters: [MoveTypeUse]
+  return: [MoveTypeUse]
+}
+
+type MoveValue {
+  type: MoveType!
+  data: MoveData
+
+  bcs: Bytes
+}
+
+union MoveData =
+    MoveNumber
+  | MoveBool
+  | MoveAddress
+  | MoveUID
+  | MoveString
+  | MoveVector
+  | MoveOption
+  | MoveStruct
+
+type MoveNumber  { number: String! }
+type MoveBool    { bool: Boolean! }
+type MoveAddress { address: SuiAddress! }
+type MoveUID     { uid: SuiAddress! }
+type MoveString  { string: String! }
+type MoveVector  { elements: [MoveValue!]! }
+type MoveOption  { element: MoveValue }
+type MoveStruct  { fields: [MoveField!]! }
+
+type MoveField {
+  name: String!
+  value: MoveValue!
+}
+
+type MoveTypeName {
+  # Fully qualified type name.  Primitive types have no `moduleId`, or
+  # `struct`.
+  moduleId: MoveModuleId
+  name: String!
+  struct: NormalizedStruct
+}
+
+type MoveType {
+  # Scalar representation of the type instantiation (type and type
+  # parameters)
+  repr: TypeInst!
+  typeName: MoveTypeName!
+  typeParameters: [MoveType]
+}
+
+type MoveTypeUse {
+  # Scalar representation of the use of a type in a signature.
+  repr: TypeSignature!
```

Comment:
`MoveTypeUse` and `TypeSignature` are not great names for what these are -- they are type instantiations that may not be fully concrete -- but I couldn't come up with a better name, suggestions welcome.